crates/nu-utils/src/filesystem.rs

```diff
@@ -83,6 +88,35 @@ pub fn have_permission(dir: impl AsRef<Path>) -> PermissionResult<'static> {
     }
 }
 
+#[cfg(all(unix, target_os = "android"))]
+pub fn have_permission(dir: impl AsRef<Path>) -> PermissionResult<'static> {
+    // on android checking access to a directory by userid and group is not
+    // enough, it is also granted by android NDK, so to check if a user has
+    // permission just cd to that directory and check
+    let old_pwd = std::env::current_dir();
```

Comment:
Environment is scoped in nushell, and multiple threads may be running, each with their own PWD. Rather than `std::env::current_dir`, `EngineState::cwd` should be used instead.